apix-impl/src/main/java/eu/xenit/apix/alfresco/search/SearchFacetsServiceImpl.java

```diff
@@ -38,16 +38,25 @@
 import org.alfresco.util.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.InitializingBean;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 // Note: To *temporarily* get working syntax highlighting and IDE features in this file: change the root
 // build.gradle to have alfresco_4_version = "5.0.d" rather than "4.2.f". Change it back before committing.
 @Component
-public class SearchFacetsServiceImpl implements SearchFacetsService {
+public class SearchFacetsServiceImpl implements SearchFacetsService, BeanFactoryAware, InitializingBean {
 
     private static final Logger logger = LoggerFactory.getLogger(SearchFacetsServiceImpl.class);
-    private final FacetLabelDisplayHandlerRegistry facetLabelDisplayHandlerRegistry;
+
+    // Beanfactory added to replace alfresco's ServiceRegistry.getFacetLabelDisplayHandlerRegistry
+    // (Due to alfresco planning to remove the class from the public api)
+    private BeanFactory beanFactory;
```

Comment:
Why fetching it from the `BeanFactory` instead of autowiring them in the constructor?

The classes and beans are still there, even when they are not public API, you can still autowire them.